hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3529,18 +3534,41 @@ func (r *HostedClusterReconciler) validateAWSConfig(hc *hyperv1.HostedCluster) e
 
 	if hc.Spec.Platform.AWS.EndpointAccess == hyperv1.Private {
 		if servicePublishingStrategy != nil && servicePublishingStrategy.Type != hyperv1.Route && servicePublishingStrategy.Type != hyperv1.LoadBalancer {
-			errs = append(errs, fmt.Errorf("service type %v with publishing strategy %v", hyperv1.APIServer, servicePublishingStrategy.Type))
+			errs = append(errs, fmt.Errorf("service type %v with publishing strategy %v is not supported, use Route", hyperv1.APIServer, servicePublishingStrategy.Type))
 		}
 
 	} else {
 		if !hyperutil.UseDedicatedDNSForKASByHC(hc) && servicePublishingStrategy.Type != hyperv1.LoadBalancer {
-			errs = append(errs, fmt.Errorf("service type %v with publishing strategy %v is not supported, use Route", hyperv1.APIServer, servicePublishingStrategy.Type))
+			errs = append(errs, fmt.Errorf("service type %v with publishing strategy %v is not supported, use Route or Loadbalancer", hyperv1.APIServer, servicePublishingStrategy.Type))
 		}
 	}
 
 	return utilerrors.NewAggregate(errs)
 }
 
+func (r *HostedClusterReconciler) validatePublishingStrategyMapping(hc *hyperv1.HostedCluster) error {
+	hostnameServiceMap := make(map[string]string, len(hc.Spec.Services))
+	for _, svc := range hc.Spec.Services {
+		hostname := ""
+		if svc.Type == hyperv1.LoadBalancer && svc.LoadBalancer != nil {
+			hostname = svc.LoadBalancer.Hostname
+		} else if svc.Type == hyperv1.Route && svc.Route != nil {
+			hostname = svc.Route.Hostname
+		}
+
+		if hostname != "" {
```

Comment:
if you invert this
```
if hostname == "" {
  continue
}
```
you can unindent the remaining lines and make it easy to understand i.e. if `hostname` is not specified, this time through the loop is a no-op.